model/map/src/main/java/org/keycloak/models/map/authorization/MapResourceServerStore.java

```diff
@@ -77,11 +76,11 @@ public ResourceServer create(String clientId) {
             throw new ModelException("Creating resource server from federated ClientModel not supported");
         }
 
-        if (tx.read(clientId) != null) {
+        if (tx.read(resourceServerStore.getKeyConvertor().fromString(clientId)) != null) {
```

Comment:
`resourceServerStore.getKeyConvertor().fromString(clientId)` can be called only once in this method.